sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/orc/OrcPartitionReaderFactory.scala

```diff
@@ -60,6 +60,7 @@ case class OrcPartitionReaderFactory(
   private val capacity = sqlConf.orcVectorizedReaderBatchSize
   private val orcFilterPushDown = sqlConf.orcFilterPushDown
   private val ignoreCorruptFiles = sqlConf.ignoreCorruptFiles
+  private val metaCacheEnabled = sqlConf.fileMetaCacheEnabled
```

Comment:
If `useFileMetaCacheList` config is used, without change the V2 API, it seems that it can only be hard coded as
```
val metaCacheEnabled = useFileMetaCacheList.toLowerCase(Locale.ROOT)
      .split(",").map(_.trim).contains("ORC")
```

the `formatName("ORC")` is defined `OrcTable`, we can't get it through the API here at present.

Reply:
7153d2a change to use a list config